src/java.base/share/classes/javax/security/auth/callback/TextInputCallback.java

```diff
@@ -43,7 +43,7 @@ public class TextInputCallback implements Callback, java.io.Serializable {
      * @serial
      * @since 1.4
      */
-    private String prompt;
+    private final String prompt;
```

Comment:
I think you can also mark `defaultText` final.

Reply:
Same as above.

Reply:
Same comment as above about setting it to a default value in the constructors.